src/xunit.performance.analysis/TestStatistics.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+#if !LINUX_BUILD
+using MathNet.Numerics.Statistics;
+#endif
+
+namespace Microsoft.Xunit.Performance.Analysis
+{
+    public class TestStatistics
+    {
+        private bool _readOnly;
+        private List<double> _values;
+        private RunningStatistics _cachedStatistics;
+
+        public TestStatistics()
+        {
+            _values = new List<double>();
+        }
+
+        public void Push(double value)
+        {
+            if (_readOnly)
+                throw new InvalidOperationException("Statistics have already been calculated.  Data is now read-only.");
+            _values.Add(value);
+        }
+
+        public RunningStatistics RunningStatistics
+        {
+            get
+            {
+                if(_cachedStatistics == null)
+                {
+                    _readOnly = true;
+                    TransformData();
+                    _cachedStatistics = new RunningStatistics(_values);
+                }
+
+                return _cachedStatistics;
+            }
+        }
+
+        public int IterationCount
+        {
+            get { return _values.Count; }
+        }
+
+        private void TransformData()
+        {
+            double[] values = _values.ToArray();
+
+            // Sort the values.
+            Array.Sort(values);
+
+            // Get the 1st and 99th percentile values.
+            double firstPercentileVal = PercentileInSortedArray(values, 1);
+            double ninetyninthPercentileVal = PercentileInSortedArray(values, 99);
+
+            int startIndex = 0;
+            int stopIndex = values.Length - 1;
+            
+            // Find the new start-index.
+            for(int i=0; i<values.Length; i++)
+            {
+                if(values[i] > firstPercentileVal)
+                {
+                    startIndex = i;
+                    break;
+                }
+            }
+
+            // Find the new end-index.
+            for(int i=values.Length-1; i>=0; i--)
+            {
+                if(values[i] < ninetyninthPercentileVal)
+                {
+                    stopIndex = i;
+                    break;
+                }
+            }
+
+            // Subset the array.
+            // Add 1 because we want to include stopIndex in the new data set.
+            int newArrayLength = stopIndex - startIndex + 1;
+            double[] newValues = new double[newArrayLength];
+            int newArrayIndex = 0;
+            for(int i=startIndex; i<=stopIndex; i++)
+            {
+                newValues[newArrayIndex++] = values[i];
```

Comment:
should use Array.public static void Copy( 	Array sourceArray, 	int sourceIndex, 	Array destinationArray, 	int destinationIndex, int length
)